src/internal/node_api_internal_types.h

```diff
@@ -114,4 +116,33 @@ struct iotjs_async_context_s {
   napi_value async_resource_name;
 };
 
+struct iotjs_tsfn_invocation_s {
+  iotjs_tsfn_invocation_t* next;
+  void* data;
+};
+
+struct iotjs_threadsafe_function_s {
+  napi_env env;
+  napi_value func;
+
+  size_t max_queue_size;
+  size_t thread_count;
+
+  void* thread_finalize_data;
+  napi_finalize thread_finalize_cb;
+  void* context;
+  napi_threadsafe_function_call_js call_js_cb;
+
+  napi_async_context async_context;
+  uv_async_t* async_handle;
+  uv_cond_t* async_cond;
+  uv_mutex_t* op_mutex;
```

Comment:
I think we could use `uv_{async/cond/mutex}_t` instead of pointers to reduce the times of malloc?